nighthawk/source/common/utility.cc

```diff
@@ -29,28 +30,105 @@ uint32_t determineCpuCoresWithAffinity() {
 
 } // namespace PlatformUtils
 
-Uri::Uri(const std::string& uri) : scheme_("http") {
+Uri Uri::Parse(absl::string_view uri) {
```

Comment:
FWIW, in a completely independent exercise, I was looking at Chromium's URL handling code, and it has all of this plus more. We're looking at how we can incorporate this Envoy-side long-term, so I feel that at some point, we can just switch to that and not have to maintain our own. Might be worth noting somewhere.